pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
-	name     	                             = "%s"
-	database  	                             = "%s"
-	schema    	                             = "%s"
-	sql_statement                            = "SELECT 1"
-	enabled  	                             = true
-	schedule                                 = "5 MINUTE"
-    user_task_managed_initial_warehouse_size = "SMALL"
-}
-`
-	return fmt.Sprintf(s, name, databaseName, schemaName)
+	rootTaskModel := model.TaskWithId("main_task", rootTaskId, true, statement).
+		WithSchedule(schedule).
+		WithSuspendTaskAfterNumFailures(2)
+	childTaskModel := model.TaskWithId("second_task", finalizerTaskId, true, statement).
+		WithFinalize(rootTaskId.FullyQualifiedName())
+	childTaskModel.SetDependsOn([]string{rootTaskModel.ResourceReference()})
+
+	firstTaskStandaloneModelDisabled := model.TaskWithId("main_task", rootTaskId, false, statement).
+		WithSchedule(schedule)
+	secondTaskStandaloneModelDisabled := model.TaskWithId("second_task", finalizerTaskId, false, statement).
+		WithSchedule(schedule)
+	secondTaskStandaloneModelDisabled.SetDependsOn([]string{firstTaskStandaloneModelDisabled.ResourceReference()})
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Task),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, firstTaskStandaloneModel) + config.FromModel(t, secondTaskStandaloneModel),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, firstTaskStandaloneModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue).
+						HasSuspendTaskAfterNumFailuresString("1"),
+					resourceshowoutputassert.TaskShowOutput(t, firstTaskStandaloneModel.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateStarted),
+					resourceassert.TaskResource(t, secondTaskStandaloneModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue),
+					resourceshowoutputassert.TaskShowOutput(t, secondTaskStandaloneModel.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateStarted),
+				),
+			},
+			// Change the second task to run after the first one (creating a DAG)
+			{
+				Config: config.FromModel(t, rootTaskModel) + config.FromModel(t, childTaskModel),
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, rootTaskModel.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanTrue).
+						HasSuspendTaskAfterNumFailuresString("2"),
+					resourceshowoutputassert.TaskShowOutput(t, rootTaskModel.ResourceReference()).
+						HasSchedule(schedule).
+						// HasTaskRelations(sdk.TaskRelations{FinalizerTask: &finalizerTaskId}).
```

Comment:
What about these comments?

Reply:
Hmm missed them somehow. Uncommented \+ added asserts to the rest of the steps.

Reply:
I left TODO for the next pr to analyze this more in-depth. For now, I don't see why it's not there. I'll take a look if that's a mapping or Snowflake issue.

Reply:
After a short analysis, It seems like a bug in Snowflake. For now, I removed this assert and added an entry in our doc about Snowflake improvements. I'll fill it later.